lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -63,31 +64,51 @@ def parse_csv(data)
       end
 
       def execute?(command, input = nil)
-        execute(command, input)
+        execute(command, :stdin => input)
         $CHILD_STATUS.success?
       end
 
-      def execute!(command, input = nil)
-        output = execute(command, input)
+      def execute!(command, options = {})
+        output = execute(command, options)
         if $CHILD_STATUS.success?
           output
         else
           raise ExecutionError.new(command, input, output)
         end
       end
 
-      def execute(command, stdin = nil)
-        logger.debug("Running command #{command} with stdin #{stdin.inspect}")
+      def execute(command, options = {})
+        stdin, hidden_patterns = parse_execute_options(options)
+        logger.debug(hide_strings("Running command #{command} with stdin #{stdin.inspect}",
+                                  hidden_patterns))
         IO.popen("#{command} 2>&1", 'r+') do |f|
           if stdin
             f.puts(stdin)
             f.close_write
           end
           output = f.read
-          logger.debug("output of the command:\n #{output}")
+          logger.debug("output of the command:\n #{hide_strings(output, hidden_patterns)}")
           output.strip
         end
       end
+
+      def parse_execute_options(options)
+        options = options.dup
+        stdin = options.delete(:stdin)
+        hidden_strings = Array(options.delete(:hidden_patterns))
+        raise ArgumentError, "Unexpected options: #{options.keys.inspect}" unless options.empty?
+        [stdin, hidden_strings]
+      end
+
+      def shellescape(string)
```

Comment:
It is not: I had it there for escaping the password on command line, but I left it there for future cases, where we will need to put user input on command line, which I expect to happen sooner or alter